### PR TITLE
Issue 545 Generalising docker-compose for multiple deployments and adding traefik as default

### PR DESCRIPTION
**Pull request by sulemanhasib43**
https://github.com/ThinkBigAnalytics/AoaCoreService/issues/545 
Generalized docker-compose.yaml for multiple deployments and add traefik overrides.

### CODE REVIEW

**Review by Christian-TD — APPROVED** [review]
LGTM. Nice work!

**Review by willfleury — COMMENTED** [review]
Instead of creating another docker-compose file specifically for traefik, can we just update the docker-compose.yaml to use traefik. We want it working by default going forward so unless there is some reason not to add it there, it would be less maintenance

File: docker-compose.yaml [hunk 1]
```diff
@@ -14,11 +14,28 @@ services:
       - '5432:5432'
 
   aoa-core:
+<<<<<<< HEAD
```
Comment (Christian-TD) [inline docker-compose.yaml]: Watch out! fix these merge conflicts.

**Review by Christian-TD — CHANGES_REQUESTED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (sulemanhasib43) [reply docker-compose.yaml]: yup fixed those. thanks man for looking out.

**Comment by sulemanhasib43** [comment]
@dartov this is tested with traefik/http and customisable env. Also thanks to @willfleury debugging s3 get recursive is also working now.

**Comment by sulemanhasib43** [comment]
we need a doc to explain .env, traefik, and how to deploy.

**Comment by sulemanhasib43** [comment]
We also need a https override .yaml for this. I will start working on it after discussion with @dartov .

**Comment by willfleury** [comment]
@sulemanhasib43 @dartov what is the story with this branch? Can we merge it? I want to do the 4.0 release and it would be better to merge this first

**Review by Christian-TD — APPROVED** [review]

**Review by Christian-TD — APPROVED** [review]

**Comment by Christian-TD** [comment]
@dartov @sulemanhasib43 please review it if you can so we can merge it. Thanks!

File: docker-compose.yaml [hunk 2]
```diff
@@ -3,91 +3,182 @@ version: "3.7"
...
     hostname: aoa-ui
     restart: always
     ports:
       - "4200:80"
     environment:
-        START_LOCAL_LOGS_SERVER: "true" 
-        LOG_MANAGEMENT_URL: "http://localhost:4200/logs/$${JOB_ID}.log"
-        BASE_URL: "http://localhost:8080/api"
-        WEBSOCKET_URL: "ws://localhost:8080/tb-ml-websocket/websocket"
-        SHOW_INFERENCE_WORKFLOW: "false"
```
Comment (dartov) [inline docker-compose.yaml:59]: Why show inference workflow by default?

File: .env [hunk 3]
```diff
@@ -0,0 +1,145 @@
+##################################
+### Analytic Ops Configuration ###
```
Comment (dartov) [inline .env:2]: typo :)

File: docker-compose.yaml [hunk 4]
```diff
@@ -3,91 +3,182 @@ version: "3.7"
...
+      DOCKER_COMPOSE_INGRESS_ENABLED: ${AOA_INGRESS_ENABLED}
+      DOCKER_NETWORK_NAME: ${AOA_DOCKER_NETWORK_NAME}
+      S3_HOST: ${AOA_S3_HOST}
+      S3_BUCKET: ${AOA_S3_BUCKET}
+      AOA_AUTOCLEAN_DOCKER: ${AOA_AUTOCLEAN_DOCKER}
+      AOA_SERVING_FQDN: ${AOA_SERVING_FQDN}
+      AOA_SERVING_SCHEME: "http"
     ports:
       - "8080:8080"
       - "61613:61613"
```
Comment (dartov) [inline docker-compose.yaml:46]: Don't need that port mapping anymore

File: websites/landing/index.html [hunk 5]
```diff
@@ -373,13 +382,14 @@ <h2 class="mdc-typography--subtitle1 weight-600">Learn</h2>
...
             "git": "https://github.com/ThinkBigAnalytics/AoaDemoModels",
-            "api": "http://localhost:8080/docs/swagger-ui.html",
+            "api": "http://api.aoa.local/docs/swagger-ui.html",
             "ci": "https://your-ci-url.com",
-            "scdlr": "http://localhost:9090/",
+            "proxy": "http://proxy.aoa.local/",
+            "scdlr": "http://scheduler.aoa.local",
             "faq": "https://apps.powerapps.com/play/4dc1574a-160d-4071-b54b-307cc7118340?tenantId=9151cbaa-fc6b-4f48-89bb-8c4a71982138?service=4",
             "docs": "https://apps.powerapps.com/play/4dc1574a-160d-4071-b54b-307cc7118340?tenantId=9151cbaa-fc6b-4f48-89bb-8c4a71982138?service=4",
             "tutorial": "http://localhost:8099/landing/help.html",
```
Comment (dartov) [inline websites/landing/index.html:395]: That needs to be http://home.aoa.local/landing/help.html

File: README.md [hunk 6]
```diff
@@ -1,16 +1,23 @@
+# Analytic Ops Docker Compose Demo Stack Deployment
```
Comment (dartov) [inline README.md:1]: Typo :)

File: README.md [hunk 7]
```diff
@@ -32,6 +39,24 @@ docker images | grep 861472724426.dkr.ecr.us-east-1.amazonaws.com/aoa
 
 You should see the images for aoa-core and aoa-ui. Keep in mind, you have to do this just once for each release or machine you are deploying to, there's no need to regularly download Core and UI images.
 
+### Setting up hosts on local machine
+
+The following command will help do this for you automatically. Note it is sudo so you will be prompted to enter your local password
+```shell
+tee -a /etc/hosts << END
+######################
+# Analytic Ops Stack #
```
Comment (dartov) [inline README.md:48]: Typo :)

**Review by dartov — APPROVED** [review]
Just minor things, I'm fine merging this, worked for me

File: docker-compose.yaml (hunk 2 above)
Comment (willfleury) [reply docker-compose.yaml:59]: I don't know why it was enabled by default but until a few commits ago on the UI it was. Saad made the change and now its disabled by default but for current 4.x release we still need this

File: README.md (hunk 7 above)
Comment (Christian-TD) [reply README.md:48]: That was my cat playing with the keyboard XD

File: docker-compose.yaml (hunk 4 above)
Comment (Christian-TD) [reply docker-compose.yaml]: nix'd

**Pull request merged**